framework/src/meshgenerators/BreakMeshByBlockGenerator.C

```diff
@@ -50,22 +51,44 @@ BreakMeshByBlockGenerator::validParams()
 BreakMeshByBlockGenerator::BreakMeshByBlockGenerator(const InputParameters & parameters)
   : BreakMeshByBlockGeneratorBase(parameters),
     _input(getMesh("input")),
-    _block(parameters.isParamSetByUser("block") ? getParam<std::vector<SubdomainID>>("block")
-                                                : std::vector<SubdomainID>()),
-    _block_set(_block.begin(), _block.end()),
-    _block_restricted(parameters.isParamSetByUser("block")),
+    _surrounding_blocks(parameters.isParamSetByUser("surrounding_blocks")
+                            ? getParam<std::vector<SubdomainID>>("surrounding_blocks")
+                            : std::vector<SubdomainID>()),
+    _block_pairs(parameters.isParamSetByUser("block_pairs")
+                     ? getParam<std::vector<std::vector<SubdomainID>>>("block_pairs")
+                     : std::vector<std::vector<SubdomainID>>()),
+    _block_pairs_restricted(parameters.isParamSetByUser("block_pairs")),
+    _surrounding_blocks_restricted(parameters.isParamSetByUser("surrounding_blocks")),
     _add_transition_interface(getParam<bool>("add_transition_interface")),
     _split_transition_interface(getParam<bool>("split_transition_interface")),
     _interface_transition_name(getParam<BoundaryName>("interface_transition_name")),
     _add_two_sides_interface(getParam<bool>("add_two_sides_interface"))
 {
+  if (_block_pairs_restricted && _surrounding_blocks_restricted)
+    paramError("'surrounding_blocks' and 'block_pairs' can not be used at the same time.");
+
+  if (_block_pairs_restricted)
+    for (auto block_pair : _block_pairs)
+    {
+      if (block_pair.size() != 2)
+        paramError("Each row of 'block_pairs' must have a size of two (block names).");
+      std::copy(block_pair.begin(), block_pair.end(), std::inserter(_block_set, _block_set.end()));
+    }
```

Comment:
I have a question here? what happens if I input 
```
block_pairs = `1 2
               2 1`
```
I believe we might need to avoid left-right duplicates.
The code should reorder the input such that the first item of the pairs is the one with the lowest id and the second the block with the highest id, and remove duplicates if presents

Reply:
Actually it does not matter whether we have duplicated items like that. In the code, we loop all the element pairs and only duplicate nodes when it can find a matching a block pair. If it finds one, it will not care whether there is another one has the same (or switch left or right) block pairs. 

I tested on the regression test by adding duplicated items. It still pass tests. I hope that means it is working fine.

Reply:
personally, I would prefer making `std::unorderedset<std::pairs<SubdomainID,SubdomainID>> _block_pairs` and then initialize it in the second part of the constructor directly using the parameter, e.g. inside the {}. Then in `findBlockPairs` you actually search for a pair inside `_block_pairs`.  Basically the same you alredy did for `_block_set`

Reply:
I do not see how it can avoid the duplication like `1 2` and `2 1`. Change to `set` only avoids duplication of a same pair, correct? Do I miss anything?

Reply:
No, you are not missing anything. When constructing `_block_pairs `  you should always put the block with the lowest id as the first element of the pair and the element with the highest id as the second element of the pair. This will also be consistent with all other routines that already work in this way.

Reply:
Ok. now I got it. That makes sense. I will make the change.